core/src/main/java/org/elasticsearch/action/index/IndexResponse.java

```diff
@@ -64,7 +74,34 @@ public String toString() {
     @Override
     public XContentBuilder innerToXContent(XContentBuilder builder, Params params) throws IOException {
         super.innerToXContent(builder, params);
-        builder.field("created", result == Result.CREATED);
+        builder.field(CREATED, result == Result.CREATED);
         return builder;
     }
+
+    /**
+     * ConstructingObjectParser used to parse the {@link IndexResponse}. We use a ObjectParser here
+     * because most fields are parsed by the parent abstract class {@link DocWriteResponse} and it's
+     * not easy to parse part of the fields in the parent class and other fields in the children class
+     * using the usual streamed parsing method.
+     */
+    private static final ConstructingObjectParser<IndexResponse, Void> PARSER;
+    static {
+        PARSER = new ConstructingObjectParser<>(IndexResponse.class.getName(),
+                args -> {
+                    // index uuid and shard id are unknown and can't be parsed back for now.
+                    ShardId shardId = new ShardId(new Index((String) args[0], IndexMetaData.INDEX_UUID_NA_VALUE), -1);
```

Comment:
We talked about this with Luca and we both think that we should not do this change. This would impact the serialization of `DocWriteResponse` in core in order to simplify things and carry less information from a client side point of view.